From 637eef1b523043f05a8089d9c74b19bce0a03dd9 Mon Sep 17 00:00:00 2001
From: Adnan Begovic <adnan@cyngn.com>
Date: Thu, 3 Mar 2016 10:56:39 -0800
Subject: [PATCH 2/2] dexdeps: Ignore static initializers on analysis.

  CTS's DexDepsXmlHandler's xml parser will error
  out when it finds a name attribute with a value
  of "<clinit>". Modify Output class to ignore static
  initializer elements to work around this.

Change-Id: I0ecbd8d0f29336cb6acba08f457bebc014c0ad5c
TICKET: CYNGNOS-2189
---
 tools/dexdeps/src/com/android/dexdeps/Output.java | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/tools/dexdeps/src/com/android/dexdeps/Output.java b/tools/dexdeps/src/com/android/dexdeps/Output.java
index dbe3bc2da..40e2592da 100644
--- a/tools/dexdeps/src/com/android/dexdeps/Output.java
+++ b/tools/dexdeps/src/com/android/dexdeps/Output.java
@@ -216,9 +216,11 @@ public class Output {
                 out.println(IN3 + "<constructor name=\"" +
                     classNameOnly(declClassName) + "\">");
             } else {
-                out.println(IN3 + "<method name=\"" + mref.getName() +
-                    "\" return=\"" + descriptorToDot(mref.getReturnTypeName()) +
-                    "\">");
+                if (!mref.getName().equals("<clinit>")) {
+                    out.println(IN3 + "<method name=\"" + mref.getName() +
+                            "\" return=\"" + descriptorToDot(mref.getReturnTypeName()) +
+                            "\">");
+                }
             }
             String[] args = mref.getArgumentTypeNames();
             for (int j = 0; j < args.length; j++) {
@@ -228,7 +230,9 @@ public class Output {
             if (constructor) {
                 out.println(IN3 + "</constructor>");
             } else {
-                out.println(IN3 + "</method>");
+                if (!mref.getName().equals("<clinit>")) {
+                    out.println(IN3 + "</method>");
+                }
             }
         }
     }
-- 
2.17.1

